artichoke-backend/src/extn/core/time/offset.rs

```diff
@@ -0,0 +1,211 @@
+                })?;
+
+                Ok(Some(offset))
+            }
+            Ruby::Float => {
+                // This impl differs from MRI. MRI supports any float value and
+                // will set an offset with subsec fractions however this is not
+                // supported in spinoso_time with the `tzrs` feature.
+                let offset_seconds: f64 = self.try_convert(in_value)?;
+
+                if (MIN_FLOAT_OFFSET..=MAX_FLOAT_OFFSET).contains(&offset_seconds) {
+                    #[allow(clippy::cast_possible_truncation)]
+                    Ok(Some(Offset::try_from(offset_seconds as i32)?))
+                } else {
+                    Err(ArgumentError::with_message("utc_offset out of range").into())
+                }
+            }
+            _ => {
+                let offset_seconds = implicitly_convert_to_int(self, in_value).and_then(|seconds| {
+                    i32::try_from(seconds).map_err(|_| ArgumentError::with_message("utc_offset out of range").into())
+                })?;
+
+                Ok(Some(Offset::try_from(offset_seconds)?))
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::extn::core::time::Offset;
+    use crate::test::prelude::*;
+
+    #[test]
+    fn no_options_does_not_raise() {
+        let mut interp = interpreter();
+
+        let options = interp.eval(b"{}").unwrap();
+
+        let offset: Option<Offset> = interp.try_convert_mut(options).unwrap();
+        assert_eq!(offset, None);
+    }
+
+    #[test]
+    fn raises_on_keys_except_in() {
+        let mut interp = interpreter();
+
+        let options = interp.eval(b"{ foo: 'bar' }").unwrap();
+
+        let result: Result<Option<Offset>, Error> = interp.try_convert_mut(options);
+        let error = result.unwrap_err();
+
+        assert_eq!(error.name(), "ArgumentError");
+        assert_eq!(error.message(), b"unknown keyword: foo".as_slice());
+    }
+
+    #[test]
+    fn raises_on_invalid_timezone_string() {
+        let mut interp = interpreter();
+
+        let options = interp.eval(b"{ in: 'J' }").unwrap();
+
+        let result: Result<Option<Offset>, Error> = interp.try_convert_mut(options);
+        let error = result.unwrap_err();
+
+        assert_eq!(error.name(), "ArgumentError");
+        assert_eq!(
+            error.message(),
+            b"\"+HH:MM\", \"-HH:MM\", \"UTC\" or \"A\"..\"I\",\"K\"..\"Z\" expected for utc_offset: J".as_slice()
+        );
+    }
+
+    #[test]
+    fn provides_an_int_based_offset() {
+        let mut interp = interpreter();
+
+        // this value is i32::MIN - 1.
+        let options = interp.eval(b"{ in: 3600 }").unwrap();
+
+        let result: Option<Offset> = interp.try_convert_mut(options).unwrap();
+        assert_eq!(result.unwrap(), Offset::fixed(3600).unwrap());
+    }
+
+    #[test]
+    fn provides_a_float_based_offset() {
+        let mut interp = interpreter();
+
+        // this value is i32::MIN - 1.
```

Comment:
It looks like this comment is a copy-paste mixup from the `raises_on_float_out_of_range` test.

```suggestion
```